app/frontend/components/options_menu.js

```diff
@@ -114,6 +115,14 @@ class OptionsMenu {
       $(this._dictionaryOption).hide();
     }
 
+    this._languageSwitcherOption = this.initConfigOption('localeSwitchOption', async () => {
+      const newLang = this._languageSwitcherOption.value;
+      console.log('Language changed', newLang);
+
+      await i18n.changeLanguage(newLang);
+      $(document).localize();
+    });
+
```

Comment:
I guess I'm not comfortable with this code. 

First, `jquery-i18next` (`$( ).localize`) still needs to be adjusted to be able to process web components. Otherwise, some kind of re-rendering needs to happen (`window.location.reload()` seems to extreme, although it does the job)

Secondly it's a part of the bigger discussion with a global state/store and dispatching events...